### PR TITLE
refactor(crypto): remove dependencies from Cargo.toml

### PR DESCRIPTION
**Pull request by mikhailOK**
Change how we access these crates - they are a part of digest crate's api

Test plan
---------
Code still compiles

### CODE REVIEW

**Comment by abacabadabacaba** [comment]
What's the point of removing dependencies from Cargo.toml if we are still using them?

**Comment by mikhailOK** [comment]
- make it explicit that we're only using them to the extent required by `digest`
- not specify their versions

**Comment by nearmax** [comment]
> What's the point of removing dependencies from Cargo.toml if we are still using them?

If we need some stuff from crate B because of crate A and crate A re-exports crate B then there is no need to depend directly on crate B. It is also not desirable because it better to let A control the version of B.

File: core/crypto/src/hash.rs [hunk 1]
```diff
@@ -1,8 +1,9 @@
 use crate::util::{Packable, Point, Scalar};
 use blake2::VarBlake2b;
+use digest::generic_array;
```
Comment (abacabadabacaba) [inline core/crypto/src/hash.rs]: Why not `use digest::generic_array::{typenum::U32, GenericArray}`?

**Pull request merged**